app/component/departure/DepartureListContainer.js

```diff
@@ -0,0 +1,193 @@
+        pattern: pattern.pattern,
+        trip: { gtfsId: stoptime.trip.gtfsId },
+      };
+    })
+  );
+
+class DepartureListContainer extends Component {
+  static propTypes = {
+    rowClasses: PropTypes.string.isRequired,
+    stoptimes: PropTypes.array.isRequired,
+    currentTime: PropTypes.object.isRequired,
+    limit: PropTypes.number,
+    infiniteScroll: PropTypes.bool,
+    showStops: PropTypes.bool,
+    routeLinks: PropTypes.bool,
+    className: PropTypes.string,
+  };
+
+  onScroll = () => {
+    if (this.props.infiniteScroll && typeof window !== 'undefined' && window !== null) {
+      return this.scrollHandler;
+    }
+    return null;
+  };
+
+  getDepartures(rowClasses) {
+    const departureObjs = [];
+    const currentTime = this.props.currentTime.unix();
+    let currentDate = this.props.currentTime.clone()
+      .startOf('day')
+      .unix();
+    let tomorrow = this.props.currentTime.clone()
+      .add(1, 'day')
+      .startOf('day')
+      .unix();
+
+    const departures = mergeDepartures(asDepartures(this.props.stoptimes))
+      .filter(departure => currentTime < departure.stoptime).slice(0, this.props.limit);
+    for (let departure of departures) {
+      if (departure.stoptime >= tomorrow) {
+        departureObjs.push(
+          <div
+            key={moment(departure.stoptime * 1000).format('DDMMYYYY')}
+            className="date-row border-bottom"
+          >
+            {moment(departure.stoptime * 1000).format('dddd D.M.YYYY')}
+          </div>);
+
+        currentDate = tomorrow;
+        tomorrow = moment.unix(currentDate)
+          .add(1, 'day')
+          .startOf('day')
+          .unix();
+      }
+
+      const id = `${departure.pattern.code} : ${departure.stoptime}`;
+
+      const validAt = alert =>
+        alert.effectiveStartDate <= departure.stoptime &&
+        departure.stoptime <= alert.effectiveEndDate &&
+        get(alert.trip.gtfsId) === get(departure.trip.gtfsId);
+
+
+      const classes = {
+        disruption: filter(departure.pattern.alerts, validAt).length > 0,
+        canceled: departure.canceled,
+      };
+
+      if (rowClasses) {
+        classes[rowClasses] = true;
+      }
+
+      if (this.props.routeLinks) {
+        departureObjs.push(
+          <Link to={`/linjat/${departure.pattern.code}`} key={id}>
+            <Departure
+              departure={departure}
+              showStop={this.props.showStops}
+              currentTime={currentTime}
+              className={cx(classes)}
+              canceled={departure.canceled}
+              isArrival={departure.isArrival}
+            />
+          </Link>);
+      } else {
+        departureObjs.push(
+          <Departure
+            key={id}
+            departure={departure}
+            showStop={this.props.showStops}
+            currentTime={currentTime}
+            className={cx(classes)}
+            canceled={departure.canceled}
+            isArrival={departure.isArrival}
+          />);
```

Comment:
Ah, no, the key prop differs. Would be better to combine shared props to hilight the differences